packages/react-instantsearch/src/core/createInstantSearchManager.js

```diff
@@ -19,16 +20,21 @@ export default function createInstantSearchManager({
   algoliaClient,
   searchParameters = {},
   resultsState,
+  stalledSearchDelay,
 }) {
   const baseSP = new SearchParameters({
     ...searchParameters,
     index: indexName,
     ...highlightTags,
   });
 
+  let stalledSearchTimer = null;
+  let isSearchStalled = false;
```

Comment:
You could store the variable `isSearchStalled` directly into the store. We avoid to keep track of the same value twice.